packages/core/src/domain/automaticErrorCollection.ts

```diff
@@ -42,32 +42,45 @@ export function filterErrors(configuration: Configuration, errorObservable: Obse
   return filteredErrorObservable
 }
 
-let originalConsoleError: (message?: any, ...optionalParams: any[]) => void
+let originalConsoleError: (...params: unknown[]) => void
 
-export function startConsoleTracking(errorObservable: ErrorObservable) {
+export function startConsoleTracking(configuration: Configuration, errorObservable: ErrorObservable) {
   originalConsoleError = console.error
-  console.error = monitor((message?: any, ...optionalParams: any[]) => {
-    originalConsoleError.apply(console, [message, ...optionalParams])
+  console.error = monitor((...params: unknown[]) => {
+    originalConsoleError.apply(console, params)
     errorObservable.notify({
-      message: ['console error:', message, ...optionalParams].map(formatConsoleParameters).join(' '),
+      ...buildErrorFromParams(configuration, params),
       source: ErrorSource.CONSOLE,
       startTime: relativeNow(),
     })
   })
 }
 
+function buildErrorFromParams(configuration: Configuration, params: unknown[]) {
+  if (configuration.isEnabled('console-stack')) {
+    const firstErrorParam = find(params, (param: unknown) => param instanceof Error) as Error
```

Comment:
```suggestion
    const firstErrorParam = find(params, (param: unknown): param is Error => param instanceof Error)
```

Reply:
that does not type `firstErrorParam` though 😕

Reply:
I would add this signature to `find` then: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/4b9438a6ae9df8e1bcaccf68d149d1643273c828/types/lodash/common/collection.d.ts#L291

```ts
export function find<T, S extends T>(
  array: T[],
  predicate: (value: T, index: number, array: T[]) => value is S
): S | undefined

```

Reply:
good reflex to check lodash!